### PR TITLE
Rename `SLP` to `SLP_MASK`

### LINKED ISSUE #917: Rename `SLP` to `SLP_MASK`
This should make it more clear that `SLP` is not a bit flag.

### PR DESCRIPTION
**Pull request by vulcandth**
Resolves #917 

Pokegold PR: https://github.com/pret/pokegold/pull/84

### CODE REVIEW

**Comment by Idain** [comment]
Looks good to me.

**Comment by Rangi42** [comment]
Thanks, this looks good! It will need doing for red+yellow too.

File: docs/bugs_and_glitches.md [hunk 1]
```diff
@@ -1141,14 +1141,14 @@ This can occur if your party and current PC box are both full when you start the
 **Fix:** Edit `PokeBallEffect` in [engine/items/item_effects.asm](https://github.com/pret/pokecrystal/blob/master/engine/items/item_effects.asm):
 
 ```diff
--; This routine is buggy. It was intended that SLP and FRZ provide a higher
+-; This routine is buggy. It was intended that SLP_MASK and FRZ provide a higher
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md:1144]:
```suggestion
-; This routine is buggy. It was intended that SLP and FRZ provide a higher
```

File: engine/items/item_effects.asm [hunk 2]
```diff
@@ -334,14 +334,14 @@ PokeBallEffect:
 	jr nz, .statuscheck
 	ld a, 1
 .statuscheck
-; This routine is buggy. It was intended that SLP and FRZ provide a higher
+; This routine is buggy. It was intended that SLP_MASK and FRZ provide a higher
```
Comment (Rangi42) [inline engine/items/item_effects.asm:337]:
```suggestion
; This routine is buggy. It was intended that SLP and FRZ provide a higher
```

**Pull request merged**